File: components/amp/test/test_mpu6050.js
/*
Please add this section into app.json when run this script as app.js.
This configuration is designed for haas100 edu k1.
"spl06": {
    "type": "I2C",
    "port": 1,
    "addrWidth": 7,
    "freq": 400000,
    "mode": "master",
    "devAddr": 105
}
*/

console.log('testing mpu6050...');

var mpu6050 = require('./mpu6050.js');

mpu6050.init("mpu6050");

var temp = mpu6050.get_Temperature();
console.log("mpu6050 temp is: " , temp);

var gy = mpu6050.get_Gyroscope();
console.log("mpu6050 gyro is: " , gy[0], gy[1], gy[2]);
var ac = mpu6050.get_Accelerometer();
console.log("mpu6050 acc is: " , ac[0], ac[1], ac[2]);

mpu6050.deinit();
console.log("test mpu6050 success!");

File: components/py_engine/tests/haas/HaaS100/python-apps/driver/i2c/test_mpu6050.py
"""
Testing mpu6050 python driver

The below i2c configuration is needed in your board.json.

"mpu6050": {
    "type": "I2C",
    "port": 1,
    "addrWidth": 7,
    "freq": 400000,
    "mode": "master",
    "devAddr": 105
}
"""

print("testing mpu6050...")
from mpu6050 import MPU6050

mpu6050Dev = MPU6050()
mpu6050Dev.open("mpu6050")
mpu6050Dev.init()
print("mpu6050 init finished")

temp = mpu6050Dev.get_Temperature()
print("mpu6050 temp is: " , temp)

gy = []
gy = mpu6050Dev.get_Gyroscope()
print("mpu6050 gyro is: " , gy[0], gy[1], gy[2])
ac = []
ac = mpu6050Dev.get_Accelerometer()
print("mpu6050 acc is: " , ac[0], ac[1], ac[2])

mpu6050Dev.close()
print("test mpu6050 success!")
